src/cocotb/regression.py

```diff
@@ -165,132 +218,81 @@ def __init__(self) -> None:
         package_name = os.getenv("RESULT_TESTPACKAGE", "all")
 
         self.xunit = XUnitReporter(filename=results_filename)
-
         self.xunit.add_testsuite(name=suite_name, package=package_name)
-
         self.xunit.add_property(name="random_seed", value=str(cocotb.RANDOM_SEED))
 
-    def discover_tests(
-        self, modules: Iterable[str], filters: Optional[Sequence[str]] = None
-    ) -> None:
+    def discover_tests(self, modules: Sequence[str]) -> None:
         """Discover tests in files automatically.
 
         Should be called before :meth:`start_regression` is called.
-        If no filters are given, all tests are included.
-        If one or more filters are given, only those tests which match a filter are included; the rest are excluded.
 
         Args:
-            modules: a list of module names to execute to find tests.
-            filters: regex patterns for test names.
-                A match *includes* the test.
+            modules: A list of module names to execute to find tests.
 
+        Raises:
+            RuntimeError: If no tests are found.
         """
         for module_name in modules:
             self.log.debug("Searching for tests in module %s", module_name)
             self.log.debug("Python Path: %s", ",".join(sys.path))
             self.log.debug("PWD: %s", os.getcwd())
-            __import__(module_name)
+            mod = import_module(module_name)
+
+            if not hasattr(mod, "__cocotb_tests__"):
+                raise RuntimeError(f"No tests were discovered in module: {module_name}")
```

Comment:
```suggestion
                raise RuntimeError(f"No tests were discovered in module: {module_name!r}")
```

Reply:
`No tests were discovered in module: example` vs `No tests were discovered in module: "example"`? I'm guessing this would also apply to the similar errors below?

Reply:
The error below for the multiple modules already does that, right, via the `repr()` in the `join`, or am I wrong?